pydm/data_plugins/epics_plugins/p4p_plugin_component.py

```diff
@@ -0,0 +1,185 @@
+                            raise ValueError(f'No matching signal for value: {value} with type: {type(value)}')
+                # Sometimes unchanged control variables appear to be returned with value changes, so checking against
+                # stored values to avoid sending misleading signals. Will revisit on data plugin changes.
+                elif changed_value == 'alarm.severity' and value.alarm.severity != self._severity:
+                    self._severity = value.alarm.severity
+                    self.new_severity_signal.emit(value.alarm.severity)
+                elif changed_value == 'display.precision' and value.display.precision != self._precision:
+                    self._precision = value.display.precision
+                    self.prec_signal.emit(value.display.precision)
+                elif changed_value == 'display.units' and value.display.units != self._units:
+                    self._units = value.display.units
+                    self.unit_signal.emit(value.display.units)
+                elif changed_value == 'control.limitLow' and value.control.limitLow != self._lower_ctrl_limit:
+                    self._lower_ctrl_limit = value.control.limitLow
+                    self.lower_ctrl_limit_signal.emit(value.control.limitLow)
+                elif changed_value == 'control.limitHigh' and value.control.limitHigh != self._upper_ctrl_limit:
+                    self._upper_ctrl_limit = value.control.limitHigh
+                    self.upper_ctrl_limit_signal.emit(value.control.limitHigh)
+                elif changed_value == 'valueAlarm.highAlarmLimit' and \
+                        value.valueAlarm.highAlarmLimit != self._upper_alarm_limit:
+                    self._upper_alarm_limit = value.valueAlarm.highAlarmLimit
+                    self.upper_alarm_limit_signal.emit(value.valueAlarm.highAlarmLimit)
+                elif changed_value == 'valueAlarm.lowAlarmLimit' and \
+                        value.valueAlarm.lowAlarmLimit != self._lower_alarm_limit:
+                    self._lower_alarm_limit = value.valueAlarm.lowAlarmLimit
+                    self.lower_alarm_limit_signal.emit(value.valueAlarm.lowAlarmLimit)
+                elif changed_value == 'valueAlarm.highWarningLimit' and \
+                        value.valueAlarm.highWarningLimit != self._upper_warning_limit:
+                    self._upper_warning_limit = value.valueAlarm.highWarningLimit
+                    self.upper_warning_limit_signal.emit(value.valueAlarm.highWarningLimit)
+                elif changed_value == 'valueAlarm.lowWarningLimit' and \
+                        value.valueAlarm.lowWarningLimit != self._lower_warning_limit:
+                    self._lower_warning_limit = value.valueAlarm.lowWarningLimit
+                    self.lower_warning_limit_signal.emit(value.valueAlarm.lowWarningLimit)
+
+    def put_value(self, value):
+        """ Write a value to the PV """
+        if is_read_only():
+            return
```

Comment:
Do we need to have some sort of warning here?

Reply:
Yes good idea, added here: 0c624c7a676363f25599abd3b23418f353d02435